### PR TITLE
Main Menu and Chairacter Select Screen

### PR DESCRIPTION
**Pull request by trulyronak**
Adding the UI for the chairacter select screen.

UI from figma — https://www.figma.com/file/VpA9bYx1dJrc6W4mWaCS8M/chairiot-the-game-but-its-in-figma?node-id=107%3A18
(dm me for edit perms)
@caIcutec i modified your UI slightly to enable the buttons we need atm, ff to redesign the figma accordingly

ui at work

Goals of this PR
- [x] Render the above UI
- [x] Be able to choose a character like this
- [x] Be able to see what players are currently connected to the game as well
- [x] Be able to ready up
- [x] Be able to force a start
- [x] Once all players are readied up, the game should start

This PR depends on
- #63 - for the UI Element Layout
- #60 - for the Chairs

### CODE REVIEW

**Comment by trulyronak** [comment]
note: things already on my mind that I want advice on

- let users click on the chair they want (i want to think about how we organize ui regions some more, and don't wanna add these buttons yet)
- fix bug where if you load the game and select a chair, a second client that joins and connects doesn't render the first one's chair (can't really figure out what's going on)

File: chariot-client/src/ui_state.rs [hunk 1]
```diff
@@ -177,6 +171,189 @@ impl GraphicsManager {
         }
     }
 
+    pub fn display_main_menu(&mut self) {
```
Comment (KNOXDEV) [inline chariot-client/src/ui_state.rs:174]: Good job on all of this and I'm glad it works, but the functional demon in me wishes it was a little less imperative and more declarative, if you know what I mean.

**Review by KNOXDEV — APPROVED** [review]
No tasking in my feedback, just comments.

Good work.

File: chariot-core/src/settings.rs [hunk 2]
```diff
@@ -46,7 +46,7 @@ impl Settings {
             .set_default("server_address", "127.0.0.1")?
             .set_default("ws_server_port", "0.0.0.0:2334")?
             .set_default("server_tick_ms", 30)?
-            .set_default("player_amount", 4)?
```
[the remark this answers is not in the document]
Comment (trulyronak) [reply chariot-core/src/settings.rs:49]: LOL my bad

[the remark this answers is not in the document]
Comment (trulyronak) [reply chariot-core/src/settings.rs:49]: I think I've just never gotten config.yaml to work for some reason

File: chariot-client/src/ui_state.rs (hunk 1 above)
Comment (trulyronak) [reply chariot-client/src/ui_state.rs:174]: wouldn't this be declarative though? this function name is explicitly stating what happens, not how?

Comment (KNOXDEV) [reply chariot-client/src/ui_state.rs:174]: lol thats not what I mean, I'm adding a comment on this line but I'm actually commenting on the whole file.

This is a function which changes the current state to the next state. There is a function for each transition of state. This is imperative, you are describing HOW. 

To do this declaratively, you would write *one* function to transition from any arbitrary state to another state. Then, you would simply create a declaration of each state.

File: chariot-core/src/settings.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (KNOXDEV) [reply chariot-core/src/settings.rs:49]: to add a config.yaml to chariot-client, you create the file `/chariot-client/config.yaml`. To add the config to the server, that is a SEPARATE FILE, `/chariot-server/config.yaml`

File: chariot-client/src/ui_state.rs (hunk 1 above)
Comment (trulyronak) [reply chariot-client/src/ui_state.rs:174]: so like a 

`pub fn change_screen(new_state: UIState)` kind of thing?

**Comment by KNOXDEV** [comment]
basically

Aaron James Eason
Roseville, CA
(916) 947-9450

On 5/23/2022 5:19:53 PM, Ronak Shah ***@***.***> wrote:
@trulyronak commented on this pull request.
In chariot-client/src/ui_state.rs [https://github.com/ucsd-cse125-sp22/cse125-sp22-group1/pull/65#discussion_r879955858]:
> @@ -177,6 +171,189 @@ impl GraphicsManager { } } + pub fn display_main_menu(&mut self) {
so like a
pub fn change_screen(new_state: UIState) kind of thing?

File: chariot-client/src/ui_state.rs (hunk 1 above)
Comment (KNOXDEV) [reply chariot-client/src/ui_state.rs:174]: basically

**Pull request merged**